proposals/1113_ldap-ad-web-ui/README.md

```diff
@@ -0,0 +1,207 @@
+  or another public Certificate Authority to prevent making end users click past messages about the certificate
+  being untrusted.
+- The name of the idp custom resource is currently not published to users logging in with Pinniped.
+  We plan on exposing this to indicate to users which idp they are logging in to.
+  Admins may need to update this to something more user-friendly.
+  Note: While branding is an important part of the user experience, and we may consider adding
+  the option to customize the page or add new fields (such as an IDP "display name" field), we
+  are choosing to defer this work until later. We want to get the MVP work done and into users'
+  hands and hope to hear more from the community once the MVP is completed.
+  For the MVP, we should not add new config but should remind admins that the IDP field field
+  is now displayed.
+
+To enable users to upgrade smoothly, the behavior of the Pinniped CLI when it encounters multiple possible flow options will change.
+Previously, the team had decided that the CLI should fail when there were multiple options (e.g. when it's could
+use either the `browser_authcode` flow or the `cli_password` flow). However, that behavior would break existing
+kubeconfigs once the `browser_authcode` flow was introduced to the IDP discovery doc.
+Instead we are opting to prioritize based on the order listed in the IDP discovery doc.
+Users will still have the option to override this priority with the `--upstream-identity-provider-flow` flag,
+but that flag will not be required.
+
+#### Tests
+
+Chromedriver browser based integration tests will be needed to ensure that a user can log in from a web-based app
+by entering their ldap credentials into the web page, as well as unit tests.
+
+With the pinniped cli:
+- succeeds with correct username and password
+- fails with incorrect username, shows useful but nonspecific error message
+- fails with incorrect password, shows useful but nonspecific error message
+- with tty access, prompts for username and password on the cli
+- without tty access, opens a browser
+- without tty access, if the form post fails, don't ask user to copy and paste the authcode (we already know you have no tty to paste it into...)
+Once dynamic clients are implemented:
+- fails when attempting to pass username/password as headers on requests to the authorize endpoint
+- tests of the rest of the dynamic client functionality that should be detailed as part of that proposal
+
+#### New Dependencies
+This should be kept to a very simple HTML page with minimal, clean CSS styling.
+Javascript should be avoided.
+The styling should match the [form post html page](https://github.com/vmware-tanzu/pinniped/tree/main/internal/oidc/provider/formposthtml)
+as much as possible, we should reuse some of the existing css and add to it to keep the style consistent.
+
+#### Observability Considerations
+* Logging login attempts at higher log levels.
```

Comment:
@margocrawf What did you mean by this? Is this an audit logging concern, or something else?

Reply:
Yes, I was thinking some basic logging of the requests we're making. `upstreamldap.go` already has this. I'll update that to say something along those lines.